app/services/reports/usps_report.rb

```diff
@@ -0,0 +1,42 @@
+require 'login_gov/hostdata'
+
+module Reports
+  class UspsReport < BaseReport
+    REPORT_NAME = 'usps-report'.freeze
+
+    def initialize
+      @results = {}
+      @results[:today] = Time.zone.today.to_s
+      @results[:letters_sent_since_days] = {}
+      @results[:letters_sent_and_validated_since_days] = {}
+      @results[:percent_sent_and_validated_since_days] = {}
+    end
+
+    def call
+      create_reports
+      save_report(REPORT_NAME, @results.to_json)
+      @results.to_json
+    end
+
+    private
+
+    def create_reports
+      [7, 14, 30, 60, 90, 10_000].each do |days_ago|
+        sent = letters_sent_since(days_ago)
+        validated = letters_sent_and_validated_since(days_ago)
```

Comment:
this is 6 x 2 = 12 table scans of the `usps_confirmation_codes` table, would it make sense to see if we can parameterize some of this and make one aggregate query? It might be some gnarly `CASE` statements

Reply:
i could do a one pass but the usps_confirmation_codes table currently has a few thousand rows.  it would need to get several orders of magnitude bigger before it ever took longer than a sub second total.